gradio/blocks.py

```diff
@@ -525,12 +530,18 @@ def __init__(
                 "mode": self.mode,
                 "custom_css": self.css is not None,
                 "theme": self.theme,
-                "version": (pkgutil.get_data(__name__, "version.txt") or b"")
-                .decode("ascii")
-                .strip(),
+                "version": GRADIO_VERSION,
             }
             utils.initiated_analytics(data)
 
+            # huggingface_hub send_telemetry
+            send_telemetry(
```

Comment:
Is this method robust? Like if there's no network, will it raise an error?

Reply:
Also, is this a blocking function? If so, we should make this run in a background thread, like we currently do in `utils.initiated_analytics()` for example